File: composeApp/src/commonMain/kotlin/com/kakapo/sudokukmp/screen/board/component/controller/ActionRowView.kt
package com.kakapo.sudokukmp.screen.board.component.controller

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.kakapo.sudokukmp.presentation.game.GameAction

@Composable
internal fun ActionRowView() {
    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceAround) {
        GameAction.entries.forEach { action ->
            val (icon, title) = action.asResource()
            ActionButtonView(image = icon, title = title, onClick = {})
        }
    }
}

File: iosApp/iosApp/screen/board/component/controller/ActionRowView.swift
import SwiftUI
import Shared

struct ActionRowView: View {
    var body: some View {
        HStack(spacing: 16) {
            ForEach(GameAction.entries, id: \.self) { action in
                ActionButtonView(action: action, onClick: {})
            }
        }
    }
}

#Preview {
    ActionRowView()
}
